src/main/java/com/exadel/discountwebapp/discount/repository/DiscountRepository.java

```diff
@@ -16,4 +17,8 @@
     @Query(value = "SELECT new com.exadel.discountwebapp.statistics.dto.DiscountViewingDTO(d.id, d.title, d.viewNumber)" +
             " FROM Discount d")
     List<DiscountViewingDTO> getDiscountSummary();
+
+    @Query(value = "SELECT new com.exadel.discountwebapp.statistics.dto.extendeddto.ExtendedDiscountViewsDTO(d.title, d.shortDescription, d.description, d.promocode, d.percentage, d.flatAmount, d.startDate, d.expirationDate, d.vendor.title, d.category.title, d.viewNumber)" +
```

Comment:
let's remove custom projection

Reply:
I try did it but where was issue (Failed to convert from type [java.lang.Object[]] to type [@org.springframework.data.jpa.repository.Query com.exadel.discountwebapp.discount.entity.Discount]), maybe cuz some data from this query not fit to Discount entity. maybe through additional fields such as (d.vendor.title, d.category.title). So I left the previous version